### PR TITLE
allowing Material almagamation without decay

### PR DESCRIPTION
**Pull request by bam241**
- Defines a PYNE_DECAY variable to detect if `decay.ccp` is part of amalgamation

- Regroups Material methods in `materials.cpp` & `material.h` to "disable" them is decay.h is not part of the amalgamated `PyNE`

- additional clean fix (clearing the comp when loading a new composition) -- I can do a separate PR for this, if you'd like

### CODE REVIEW

**Comment by gonuke** [comment]
If you rebase this against changes in `material.cpp` I can review

**Comment by bam241** [comment]
I thought I did... but my local upstream/master was not up-to-date.... Sorry about that

**Comment by bam241** [comment]
@gonuke merge done.

File: src/material.cpp [hunk 1]
```diff
@@ -87,7 +90,7 @@ void pyne::Material::_load_comp_protocol1(hid_t db, std::string datapath,
   hid_t data_set = H5Dopen2(db, datapath.c_str(), H5P_DEFAULT);
 
   // Grab the nucpath
-  if (detect_nuclidelist(data_set, nucpath)){
+  If (detect_nuclidelist(data_set, nucpath)){
```
Comment (gonuke) [inline src/material.cpp]: oops!
```suggestion
  if (detect_nuclidelist(data_set, nucpath)){
```

File: src/material.cpp [hunk 2]
```diff
@@ -2027,6 +1973,65 @@ pyne::Material pyne::Material::cram(std::vector<double> A,
...
+                       decay_const(i->first) * inhale_dose(i->first, source) / \
+                       atomic_mass(i->first);
+    }
+  } else {
+    throw std::invalid_argument("Dose type must be one of: ext_air, ext_soil, ingest, inhale.");
+  }
+  return dose;
+}
+
+#endif
```
Comment (gonuke) [inline src/material.cpp]: I think this syntax is OK, and it helps the reader find which `#if` statement this is ending.

```suggestion
#endif  PYNE_DECAY
```

File: src/material.h [hunk 3]
```diff
@@ -422,7 +406,24 @@ namespace pyne
     /// Returns a mapping of the nuclides in this material to their atom densities.
     /// This calculation is based off of the material's density.
     std::map<int, double> to_atom_dens();
-
+#ifdef PYNE_DECAY
+    /// Calculates the activity of a material based on the composition and each
+    /// nuclide's mass, decay_const, and atmoic_mass.
+    comp_map activity();
```
Comment (gonuke) [inline src/material.h]: None of these rely on decay, do they?  Just on decay data which is found elsewhere.  Can't we keep these even if decay is not included?

File: src/material.cpp [hunk 4]
```diff
@@ -45,6 +45,9 @@ void pyne::Material::norm_comp() {
 
 
 void pyne::Material::_load_comp_protocol0(hid_t db, std::string datapath, int row) {
+  // Clear current content
+  comp.clear();
```
Comment (gonuke) [inline src/material.cpp]: Is this related to the other changes in this PR????

**Review by gonuke — CHANGES_REQUESTED** [review]
Oops - didn't include my changes in a review.

I have a few suggestions/requests.

File: src/material.h (hunk 3 above)
Comment (bam241) [reply src/material.h]: we surely can

Comment (bam241) [reply src/material.h]: this is correct now only include `Material::decay()` in the `IFDEF`

File: src/material.cpp (hunk 4 above)
Comment (bam241) [reply src/material.cpp]: some merge left over.

**Comment by gonuke** [comment]
Moving those methods in & out of the DECAY option has made this look like a much bigger change than it is because a bunch of code is now just moved around in the file with no changes.  Can you streamline this to only include the changes that matter?

**Comment by bam241** [comment]
> Moving those methods in & out of the DECAY option has made this look like a much bigger change than it is because a bunch of code is now just moved around in the file with no changes. Can you streamline this to only include the changes that matter?

This should be clean now :)

**Comment by gonuke** [comment]
Thanks @bam241

**Review by gonuke — APPROVED** [review]
